docs/milestone-1.md

```diff
@@ -0,0 +1,175 @@
+6. the BN must use the [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayloadheader) received to assemble and sign a [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock) and return it to mev-boost using [`builder_proposeBlindedBlockV1`](#builder_proposeblindedblockv1).
+7. mev-boost must forward the [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock) to all connected relays and attach the matching [`SignedMEVPayloadHeader`](#signedmevpayloadheader) using [`relay_proposeBlindedBlockV1`](#relay_proposeblindedblockv1) to inform the network of which relay created this payload.
+8. if an [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayload) is returned, mev-boost must verify that the root of the transaction list matches the expected transaction root from the [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock) before returning it to the BN.
+
+### required client modifications
+
+- consensus client must implement [blind transaction signing](https://hackmd.io/@paulhauner/H1XifIQ_t#Change-1-Blind-Transaction-Signing)
+
+## API Docs
+
+Methods are prefixed using the following convention:
+- `engine` prefix indicates calls made to the execution client. These methods are specified in the [execution engine APIs](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md).
+- `builder` prefix indicates calls made to the mev-boost middleware.
+- `relay` prefix indicates calls made to a relay.
+
+### engine_forkchoiceUpdatedV1
+
+See [engine_forkchoiceUpdatedV1](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#engine_forkchoiceupdatedv1).
+
+### engine_getPayloadV1
+
+See [engine_getPayloadV1](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.5/src/engine/specification.md#engine_getpayloadv1).
+
+### builder_getPayloadHeaderV1
+
+#### Request
+
+- method: `builder_getPayloadHeaderV1`
+- params:
+  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+
+#### Response
+
+- result: [`ExecutionPayloadHeaderV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.6/specs/merge/beacon-chain.md#executionpayloadheader)
+- error: code and message set in case an exception happens while getting the payload.
+
+### builder_proposeBlindedBlockV1
+
+#### Request
+
+- method: `builder_proposeBlindedBlockV1`
+- params:
+  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
```

Comment:
This is JSON-RPC serialized, while `SignedBlindedBeaconBlock` is only specified here as an SSZ type. Aspects of SSZ type serialization aren't otherwise specified by the engine API spec on which this builds, such as `SignedBlindedBeaconBlock.signature` and `SignedMEVPayloadHeader.signature` (both `BLSSignature`s), leaving room for noninteroperability across implementations.

https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.6/src/engine/specification.md#executionpayloadv1 provides an explicit mapping between a JSON-RPC-serialization of `engine.ExecutionPayloadV1` and SSZ representation of `bellatrix.ExecutionPayload`, where both are supposed to be otherwise isomorphic.

Reply:
I agree this should be in JSON-RPC terms.  

We could fully specify blocks here, similar to how `ExecutionPayloadV1` is defined by the execution API, but that might not be worth doing and maintaining because we already have blocks specified [here](https://github.com/ethereum/beacon-APIs/blob/master/types/altair/block.yaml), and I don't think we'd want to leave room for discrepancy between JSON serialization of the same type between the two APIs.

I've raised a PR [here](https://github.com/ethereum/beacon-APIs/pull/194) to add Bellatrix types to the beacon API as well as endpoints and types necessary for blind transaction signing.

Reply:
Thanks @tersec and @realbigsean. I need help here to understand the proposal of using JSON-RPC for encoding these new methods. As I understand, SSZ is the new encoding for the consensus layer. I wonder why do we want to use the old encoding for the new things. Probably this has been discussed long ago, so please be patient with me :)
The way I imagine it, these APIs will be implemented by new software in the future, so these seems like a good way to start moving the future to be more consistent and easier to read. Am I being naive or missing some part of the story?

Reply:
SSZ is used as as the intra-consensus layer/inter-beacon-node serialization over the libp2p gossip network, yes.

JSON-RPC serializations appear in the REST API and related areas, e.g., validator clients mostly use JSON-RPC-type encodings to interact with beacon nodes. Both remain in use.